plugins/inputs/mysql/mysql.go

```diff
@@ -641,11 +658,24 @@ func (m *Mysql) gatherSlaveStatuses(db *sql.DB, serv string, acc telegraf.Accumu
 			if m.MetricVersion >= 2 {
 				col = strings.ToLower(col)
 			}
-			if value, ok := m.parseValue(*vals[i].(*sql.RawBytes)); ok {
+
+			if m.GatherAllSlaveChannels &&
+				(strings.ToLower(col) == "channel_name" || strings.ToLower(col) == "connection_name") {
+				// Since the default channel name is empty, we need this block
+				channelName := "default"
+				if len(*vals[i].(*sql.RawBytes)) > 0 {
+					channelName = string(*vals[i].(*sql.RawBytes))
+				}
+				tags["channel"] = channelName
+			} else if value, ok := m.parseValue(*vals[i].(*sql.RawBytes)); ok {
 				fields["slave_"+col] = value
 			}
 		}
 		acc.AddFields("mysql", fields, tags)
+
+		if !m.GatherAllSlaveChannels {
```

Comment:
It seems like `break`ing early would disturb current functionality. Since the loop logic is the same as before, we should expect someone is parsing subsequent rows.

Reply:
Breaking early is what we have right now (with if instead of while) when we do not check all channels. So, this if does the same early break.

Reply:
Can you please add a small comment here saying something like
```golang
// Only the first row is relevant if not all slave-channels should be gathered, so break here and skip the remaining rows
```

Reply:
Added the comment